proposals/VBnet-Coding-Guidelines.md

```diff
@@ -0,0 +1,78 @@
+# VB.net Coding Guidelines
+ 
+ Coding Conventions
+ 
+ * 80 character line limit    
+ Is the recommended maximum length of single line of code.    
+*As seen in a text editor, not in the context of the actual language specification*
+
+ * Use 4 space indentation, not tabs.
+ 
+ * Use plain code to validation parameters at public boundaries.    
+   * Do not use Contracts or magic helpers.
+   * Use `Debug.Assert()` for checks not needed in release builds. Always include a contextually meaningful message string in your assert to identify failure conditions.
+   * Add assertions to documentation assumptions on non-local program state or parameter values, eg.    
+   	*"At this point in parsing the scanner should have been advanced to a '.' token by the caller".*
+   * Avoid allocations in compiler hot paths.
```

Comment:
:memo: This is nested under "Use plain code to validat[e] parameters..." but doesn't appear related